coderd/initscript_test.go

```diff
@@ -0,0 +1,45 @@
+package coderd_test
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/v2/coderd/coderdtest"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+func TestInitScript(t *testing.T) {
+	t.Parallel()
+
+	t.Run("OK Windows", func(t *testing.T) {
+		t.Parallel()
+		client := coderdtest.New(t, nil)
+		script, err := client.InitScript(context.Background(), "windows", "amd64")
+		require.NoError(t, err)
+		require.NotEmpty(t, script)
+		require.Contains(t, script, "$env:CODER_AGENT_AUTH = \"token\"")
+	})
+
+	t.Run("OK Linux", func(t *testing.T) {
+		t.Parallel()
+		client := coderdtest.New(t, nil)
+		script, err := client.InitScript(context.Background(), "linux", "amd64")
```

Comment:
Probably should make one of these `arm64` for variety, and also assert that the string `/bin/coder-OS-ARCH[.exe]` shows up